packages/flutter_tools/lib/src/build_system/targets/common.dart

```diff
@@ -255,6 +256,7 @@ class KernelSnapshot extends Target {
     final String dillPath = environment.buildDir.childFile(dillName).path;
 
     final List<String> dartDefines = decodeDartDefines(environment.defines, kDartDefines);
+    _addFeatureFlagsToDartDefines(dartDefines);
```

Comment:
I wonder if this would work in tests.

That is, are flags picked up by:
```dart
flutter test test/foo_test.dart
```

It's probably OK if the answer is no.

Reply:
Yup, this works for unit tests, widget tests, and integration tests